### DISCUSSION TITLE
How to push Gist?

### DISCUSSION THREAD

**Discussion by marcelomgarcia** [community/community#62203] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Dear all,

I cloned a Gist:

```
git clone https://gist.github.com/b9186eed2601dea0c16b8a0dc2e8b4ec.git
```

After editing the file, when I push to update the Gist, I get an error:

```
git push
remote: Repository not found.
fatal: repository 'https://gist.github.com/b9186eed2601dea0c16b8a0dc2e8b4ec.git/' not found
```

I thought the problem was my _username_ missing from the URL, so i tried to update the URL 

```
git remote set-url origin https://gist.github.com/mar...arcia/b9186eed2601dea0c16b8a0dc2e8b4ec
```

But it made no difference. 

How to update (push) to a Gist?

**Comment by Beargruug** [comment, marked as answer]
Hey,

When you clone a Gist using Git, you are essentially creating a read-only copy of the Gist content on your local machine. You can make changes to the files locally, but there is no direct way to push those changes back to the original Gist through Git.

If you want to update the content of a Gist, you can do so by visiting the Gist page on the GitHub website, making the changes there, and then clicking the "Update gist" button to save the changes. You'll need to be logged in to your GitHub account to be able to update the Gist.

Hope this helps!

**Comment by nirav-gajera** [comment]
The error you are getting is because you are trying to push to a public Gist, which you cannot do. Public Gists are read-only, so you cannot push changes to them.

If you want to update a public Gist, you need to fork it first. Forking a Gist creates a copy of it in your own GitHub account, which you can then push changes to.

To fork a Gist, follow these steps:

1. Go to the Gist you want to fork.
2. Click the "Fork" button.
3. Sign in to your GitHub account if you are not already signed in.
4. The Gist will be forked to your account.

Once you have forked the Gist, you can make changes to it and push them to GitHub. To do this, follow these steps:

1. Clone the forked Gist to your local machine.
2. Make changes to the files.
3. Commit the changes.
4. Push the changes to GitHub.

Here are the commands you need to run:

```
git clone https://gist.github.com/<username>/<gist_id>.git
git add .
git commit -m "Your commit message"
git push origin master
```

Replace `<username>` with your GitHub username and `<gist_id>` with the ID of the Gist you forked.

**Comment by marcelomgarcia** [comment]
Hi, 

I didn't understand that Gists were a read-only repository. I will update directly on the GitHub page.

Thank you.

**Comment by marcelomgarcia** [comment]
Closing ticket

**Reply by tuves** [reply]
Hi @marcelomgarcia if someone has answered your question, please remember to mark them as answered (I've done it for you above). This will help other members with the same question find the solution faster, as well as give credit to the members who have taken time to answer your question 😃

**Reply by marcelomgarcia** [reply]
Hi, I tried to close this ticket twice (the previous comments). I didn't understand how to close: I chose the option to close, them I clicked on the "reply" button, and in the second time, just in "comment and close(?)". Apparently none of them worked. Thanks for closing the ticket.

**Comment by sinetris** [comment]
Hi @marcelomgarcia 
I don't know if it's a new feature (I don't remember if I tried before 😅), but I can push to a gist **I** created if I clone them using "Clone via SSH".
<img width="488" alt="gist-clone-via-ssh" src="https://github.com/community/community/assets/191281/fd6f933e-9d62-4c75-8c61-6bb7aab26257">

```shell
gist_hash="<the-gist-hash>"
gist_local_dir="<a-better-name-for-the-local-gist>"
git clone git@gist.github.com:${gist_hash}.git "${gist_local_dir}"
cd "${gist_local_dir}"
# change files + git add + git commit
git push
# there should be a new revision in your gist under *revisions*
```

> [!IMPORTANT]
> Gist does not support directories.

You will get an error from a pre-hook if you try to push a commit that includes directories (so nothing will change in your gist).

**Reply by yarikoptic** [reply]
Thank you!  Works nicely. Moreover I added to my `~/.gitconfig`

```
[url "git@gist.github.com:"]
    pushinsteadOf = https://gist.github.com/yarikoptic/
```

where `yarikoptic` is my handle on github. Now I can `git clone https://gist.github.com/yarikoptic/314f9f419c7203bcc40331571cacf693` and then still `git push` (via ssh). I heavily rely on similar setup also for the main github.com so can clone directly via https (no authentication) but keep pushing via ssh whenever needed without changing any config.

**Reply by rivenirvana** [reply]
Just an update which may be helpful to those that find themselves here.

The instructed `pushinsteadOf` value wasn't working for me, and it seems like your username isn't needed anymore. I just left it as `pushinsteadOf = https://gist.github.com/` and restored intended behavior of https fetch -> ssh push.

**Reply by victorlin** [reply]
I still need to include username. Without username, I get `ERROR: Repository not found`.

**Reply by 4ft35t** [reply]
Great，works fine.

**Comment by chipbite** [comment]
Is there a bug report / feature request to add support for pushing updates to a gist via https?

It would allow us on ssh-disabled-corporate-networks to work with gists more effectively.

**Comment by wangshuai-007** [comment]
If you get stuck in `git push` command.

You can try `ssh -T git@gist.github.com` test connect with `github`.

Then ,printe the log to get more detail:`ssh -vvvT -o ConnectTimeout=10 git@gist.github.com`

Then copy output to ask AI to solve the problem
